plugin-infra/go-plugin-infra/src/com/thoughtworks/go/plugin/infra/commons/PluginsZip.java

```diff
@@ -73,7 +74,21 @@ private String computeMd5DigestOfPlugins() {
             String digestOfPlugins = digestOfBundledFolder + digestOfExternalFolder;
             return DigestUtils.md5Hex(digestOfPlugins);
         } catch (Exception e) {
-            throw new RuntimeException(String.format("Could not compute md5 of plugins. Exception occurred: %s", e.getStackTrace()));
+            throw new RuntimeException(String.format("Could not compute md5 of plugins." +
+                    " Exception occurred: %s", e.getStackTrace()));
         }
     }
+
+    private void checkFilesAccessibility(File bundledPlugins, File externalPlugins) {
+        try {
+            if (!bundledPlugins.canRead() || !externalPlugins.canRead()) {
+                LOG.error("Could not read plugins." +
+                        "Please check access rights on files in external and bundled folder.");
+                throw new FileAccessRightsCheckException("Could not read plugins. " +
+                        "Please check access rights in external and bundled plugin folder.");
+            }
+        } catch (SecurityException se) {
```

Comment:
Are you trying to catch this from `File#canRead()`?

We aren't catching `SecurityException` anywhere in the app, I feel it's OK to drop this catch block.